fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java

```diff
@@ -170,4 +191,42 @@ public ExecPlan plan(UpdateStmt updateStmt, ConnectContext session) {
             }
         }
     }
+
+    /**
+     * Cast output columns type to target type.
+     * @param columnRefFactory :  column ref factory of update stmt.
+     * @param targetTable: target table of update stmt.
+     * @param colNames: column names of update stmt.
+     * @param outputColumns: output columns of update stmt.
+     * @param root: root logical plan of update stmt.
+     * @return: new root logical plan with cast operator.
+     */
+    private static OptExprBuilder castOutputColumnsTypeToTargetColumns(ColumnRefFactory columnRefFactory,
+                                                                      Table targetTable,
+                                                                      List<String> colNames,
+                                                                      List<ColumnRefOperator> outputColumns,
+                                                                      OptExprBuilder root) {
+        Map<ColumnRefOperator, ScalarOperator> columnRefMap = new HashMap<>();
+        ScalarOperatorRewriter rewriter = new ScalarOperatorRewriter();
+        List<ScalarOperatorRewriteRule> rewriteRules = Arrays.asList(new FoldConstantsRule());
+        Preconditions.checkState(colNames.size() == outputColumns.size(), "Column name's size %s should be equal " +
+                        "to output column refs' size %s", colNames.size(), outputColumns.size());
+
+        for (int columnIdx = 0; columnIdx < outputColumns.size(); ++columnIdx) {
+            ColumnRefOperator outputColumn = outputColumns.get(columnIdx);
+            String colName = colNames.get(columnIdx);
+            Column column = targetTable.getColumn(colName);
+            Preconditions.checkState(column != null, "Column %s not found in table %s", colName,
+                    targetTable.getName());
+            if (!column.getType().matchesType(outputColumn.getType())) {
```

Comment:
check the type can cast to target type？

Reply:
Add it. But this should not happen since it will fail in the analyzer phase.

Reply:
Add more tests.